src/lua/LuaSystemView.cpp

```diff
@@ -0,0 +1,417 @@
+
+	TransferPlanner *planner = sv->GetTransferPlanner();
+	if (key == "prograde") {
+		LuaPush<double>(l, planner->GetDv(BurnDirection::PROGRADE));
+	} else if (key == "normal") {
+		LuaPush<double>(l, planner->GetDv(BurnDirection::NORMAL));
+	} else if (key == "radial") {
+		LuaPush<double>(l, planner->GetDv(BurnDirection::RADIAL));
+	} else if (key == "starttime") {
+		LuaPush<double>(l, planner->GetStartTime());
+	} else if (key == "factor") {
+		LuaPush<double>(l, planner->GetFactor() * 10); // factor is shown as "x 10"
+	} else {
+		Warning("Unknown transfer planner key %s\n", key.c_str());
+		lua_pushnil(l);
+	}
+	return 1;
+}
+
+static int l_systemview_transfer_planner_add(lua_State *l)
+{
+	SystemView *sv = LuaObject<SystemView>::CheckFromLua(1);
+	std::string key = LuaPull<std::string>(l, 2);
+	double delta = LuaPull<double>(l, 3);
+
+	TransferPlanner *planner = sv->GetTransferPlanner();
+	if (key == "prograde") {
+		planner->AddDv(BurnDirection::PROGRADE, delta);
+	} else if (key == "normal") {
+		planner->AddDv(BurnDirection::NORMAL, delta);
+	} else if (key == "radial") {
+		planner->AddDv(BurnDirection::RADIAL, delta);
+	} else if (key == "starttime") {
+		planner->AddStartTime(delta);
+	} else if (key == "factor") {
+		if (delta > 0)
+			planner->IncreaseFactor();
+		else
+			planner->DecreaseFactor();
+	} else {
+		Warning("Unknown transfer planner key %s\n", key.c_str());
+	}
+	return 0;
+}
+
+static int l_systemview_transfer_planner_reset(lua_State *l)
+{
+	SystemView *sv = LuaObject<SystemView>::CheckFromLua(1);
+	std::string key = LuaPull<std::string>(l, 2);
+
+	TransferPlanner *planner = sv->GetTransferPlanner();
+	if (key == "prograde") {
+		planner->ResetDv(BurnDirection::PROGRADE);
+	} else if (key == "normal") {
+		planner->ResetDv(BurnDirection::NORMAL);
+	} else if (key == "radial") {
+		planner->ResetDv(BurnDirection::RADIAL);
+	} else if (key == "starttime") {
+		planner->ResetStartTime();
+	} else if (key == "factor") {
+		planner->ResetFactor();
+	} else {
+		Warning("Unknown transfer planner key %s\n", key.c_str());
+	}
+
+	return 0;
+}
+
+template <>
+const char *LuaObject<SystemView>::s_type = "SystemView";
+
+template <>
+void LuaObject<SystemView>::RegisterClass()
+{
+	static const luaL_Reg l_methods[] = {
+
+		{ "GetProjectedGrouped", l_systemview_get_projected_grouped },
+		{ "GetSelectedObject", l_systemview_get_selected_object },
+		{ "GetOrbitPlannerStartTime", l_systemview_get_orbit_planner_start_time },
+		{ "GetOrbitPlannerTime", l_systemview_get_orbit_planner_time },
+		{ "AccelerateTime", l_systemview_accelerate_time },
+		{ "SetSelectedObject", l_systemview_set_selected_object },
+		{ "SetVisibility", l_systemview_set_visibility },
+		{ "SetColor", l_systemview_set_color },
+
+		{ "TransferPlannerAdd", l_systemview_transfer_planner_add },
+		{ "TransferPlannerGet", l_systemview_transfer_planner_get },
+		{ "TransferPlannerReset", l_systemview_transfer_planner_reset }
```

Comment:
You're missing a `{ NULL, NULL }` terminator at the end of this list. On my system (using AVX2-accelerated strlen()), this code crashes the entire application when the terminator is not present.

Reply:
Done in e32068af86141c1035fea259612b3dec5fed1959!